### DISCUSSION TITLE
Which tags are indexed?

### DISCUSSION THREAD

**Discussion by karussell** [komoot/photon#813] [General]
Using [this query](https://photon.komoot.io/reverse?lat=51.42919&lon=14.268279&osm_tag=railway:station&radius=5) to find railway stations I get only [this node](https://www.openstreetmap.org/node/3028759519) because the other railway station is only tagged with `public_transport=station`. But using [a query with this tag](https://photon.komoot.io/reverse?lat=51.42919&lon=14.268279&osm_tag=public_transport:station&radius=5) won't give any results. Is the problem that `public_transport` is not indexed? Is this related to https://github.com/komoot/photon/issues/462 ?

And is this also the reason why [this toilet](https://www.openstreetmap.org/way/158486039) isn't found? Here is the [photon query](https://photon.komoot.io/reverse?lat=51.4333103&lon=14.250083&radius=5&osm_tag=amenity%3Atoilets).

**Comment by lonvia** [comment]
Photon imports all data from the Nominatim database that have either a name or a housenumber, see  https://github.com/komoot/photon/blob/ce3424bf73d0dee8cf859a73ff2f2eefd745145e/src/main/java/de/komoot/photon/PhotonDoc.java#L200-L206

The Nominatim instance used to produce the pre-built Photon indexes uses a special style that drops all the OSM data that doesn't have a name.

Please note once again that Photon is mainly meant to be a forward geocoder. It is supposed to find places by their name and address. It is not meant to be a POI database. It would need to have a different design, if it was supposed to be a POI database. If you are using a query of the form `reverse?lat=..&lon=..&osm_tag=...&limit=...` then you are very close to using Photon out-of-spec. Please have a look at Overpass API.

**Reply by karussell** [reply]
Thanks for these details. 

When I use osm_tag=historic it shows me only `historic=memorial` although e.g. this tower (building=yes):

https://www.openstreetmap.org/way/244850905

has a name and an address with housenumber and it does not show up for the following requests:

https://photon.komoot.io/reverse?lat=50.850814&lon=4.349807&radius=50&limit=10&osm_tag=historic

https://photon.komoot.io/reverse?lat=50.850814&lon=4.349807&radius=50&limit=10&osm_tag=historic:tower

Are there other requirements?

Do I need to add certain "extra-tags" in the command line?

**Reply by karussell** [reply]
Fetching the ID=244850905 in ElasticSearch directly gives two documents:

 * one with "osm_key" : "man_made" and "osm_value" : "tower" and 
 * the other with "osm_key" : "tourism" and "osm_value" : "attraction".

Both with default.name = Tour Noire - Zwarte Toren.

Where is the code which tags trigger creating multiple documents? Or is this limitation coming from nominatim? (But it seems the needed tag [is stored in nominatim](https://nominatim.openstreetmap.org/ui/details.html?osmtype=W&osmid=244850905&class=man_made).)

Could we improve the situation and index all tags like:

```json
{
  "tags": [
    { "historic": "tower" },
    { "tourism": "attraction" },
    ...
  ]}
```

instead of creating multiple documents? Assuming the following mapping (or a dynamic mapping):

```json
{
  "mappings": {
    "properties": {
      "tags": {
        "type": "nested",
        "properties": {
          "historic": { "type": "keyword" },
          "tourism": { "type": "keyword" },
          ...
        }}}}}
```

the query would look like this `{ "term": { "tags.historic": "tower" } }`. Or would this explode the index?

**Reply by lonvia** [reply]
The duplication comes from Nominatim. We should get rid of it eventually but I'd prefer to reconsider this in a larger context of overhauling Nominatim's simplistic class/type classification.

**Reply by lonvia** [reply]
`historic` is an odd tag because it is sometimes a main tag, sometimes just an attribute. In Nominatim it is only used as classification when there are no other useful tags. Photon does not save the full tagging of places. It just gets a crude classification from Nominatim which happens to correspond to one of the OSM tags. You cannot use Photon to filter by arbitrary tags.
